File: 1695-maximum-erasure-value/1695-maximum-erasure-value.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumUniqueSubarray = function(nums) {
  const set = new Set();
  let [j, sum, ans] = [0, 0, 0];
  
  for (let num of nums) {
    sum += num;
    while (set.has(num)) {
      set.delete(nums[j])
      sum -= nums[j++];
    }
    
    set.add(num);
    ans = Math.max(ans, sum);
  }
  
  return ans;
};

File: 1695-maximum-erasure-value/1695-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        seen = set()
        j, s, ans = 0, 0, 0
        for num in nums:
            s += num
            while num in seen:
                seen.remove(nums[j])
                s -= nums[j]
                j += 1
            seen.add(num)
            ans = max(ans, s)
        return ans